test/nodejs-bundle/circuit/dial-over-specific-relay.js

```diff
@@ -0,0 +1,205 @@
+/* eslint-env mocha */
+'use strict'
+
+const TCP = require('libp2p-tcp')
+const WS = require('libp2p-websockets')
+const multiplex = require('libp2p-multiplex')
+
+const waterfall = require('async/waterfall')
+const utils = require('./helpers/utils')
+
+const sinon = require('sinon')
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+
+const expect = chai.expect
+
+const nodeKeys = require('./fixtures/nodes')
+const Circuit = require('libp2p-circuit')
+
+describe(`listen on an explicit relay addr`, function () {
+  this.timeout(500000)
+
+  let portBase = 9020 // TODO: randomize or mock sockets
+  let testNodes
+
+  let relaySpy1
+  let relaySpy2
+  let active = false
+
+  function setUpNodes (muxer, done) {
+    utils.createNodes(
+      {
+        relayNode1: {
+          id: nodeKeys.node1,
+          transports: [new TCP(), new WS()],
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}`,
+            `/ip4/0.0.0.0/tcp/${portBase++}/ws`
+          ],
+          isCrypto: true,
+          config: {
+            relay: {
+              circuit: {
+                enabled: true,
+                active: active
+              }
+            }
+          }
+        },
+        relayNode2: {
+          id: nodeKeys.node2,
+          transports: [new TCP(), new WS()],
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}`,
+            `/ip4/0.0.0.0/tcp/${portBase++}/ws`
+          ],
+          isCrypto: true,
+          config: {
+            relay: {
+              circuit: {
+                enabled: true,
+                active: active
+              }
+            }
+          }
+        },
+        nodeA: {
+          id: nodeKeys.node3,
+          transports: [new TCP()],
+          isCrypto: true,
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}`,
+            `/ip4/0.0.0.0/tcp/9022/ipfs/${nodeKeys.node2.id}/p2p-circuit`,
+            `/ipfs/${nodeKeys.node2.id}/p2p-circuit`
+          ]
+        },
+        nodeB: {
+          id: nodeKeys.node4,
+          transports: [new WS()],
+          isCrypto: true,
+          muxer: muxer,
+          addrs: [
+            `/ip4/0.0.0.0/tcp/${portBase++}/ws`,
+            `/ip4/0.0.0.0/tcp/9021/ws/ipfs/${nodeKeys.node1.id}/p2p-circuit`,
+            `/ipfs/${nodeKeys.node1.id}/p2p-circuit`
+          ]
+        }
+      },
+      (err, nodes) => {
+        if (err) {
+          return done(err)
+        }
+
+        testNodes = nodes
+        relaySpy1 = sinon.spy(testNodes['relayNode1'].swarm.transports[Circuit.name].listeners[0].hopHandler, 'handle')
+        relaySpy2 = sinon.spy(testNodes['relayNode2'].swarm.transports[Circuit.name].listeners[0].hopHandler, 'handle')
+
+        done()
+      })
```

Comment:
You have repeated this coded multiple times in every single test, please use the DRY principle and try to reuse what exists already, such as: https://github.com/libp2p/js-libp2p/blob/master/test/nodejs-bundle/utils.js#L12-L34